daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/NilEmptyCombinatorParsers.scala

```diff
@@ -19,14 +19,8 @@ package org.apache.daffodil.processors.parsers
 
 import org.apache.daffodil.processors.TermRuntimeData
 
-case class SimpleNilOrEmptyOrValueParser(ctxt: TermRuntimeData, nilParser: Parser, emptyParser: Parser, valueParser: Parser)
-  extends AltCompParser(ctxt, Seq(nilParser, emptyParser, valueParser))
-
 case class SimpleNilOrValueParser(ctxt: TermRuntimeData, nilParser: Parser, valueParser: Parser)
-  extends AltCompParser(ctxt, Seq(nilParser, valueParser))
-
-case class SimpleEmptyOrValueParser(ctxt: TermRuntimeData, emptyParser: Parser, valueParser: Parser)
-  extends AltCompParser(ctxt, Seq(emptyParser, valueParser))
+  extends ChoiceParser(ctxt, Seq(nilParser, valueParser))
 
 case class ComplexNilOrContentParser(ctxt: TermRuntimeData, emptyParser: Parser, contentParser: Parser)
-  extends AltCompParser(ctxt, Seq(emptyParser, contentParser))
+  extends ChoiceParser(ctxt, Seq(emptyParser, contentParser))
```

Comment:
I like this change to replace AltCompParser with ChoiceParser, they're virtually he same. One thing to consider though is that when the ChoiceParser fails it creates parser errors that reference "Choices". There's potential confusion there. The user might think the error is related to xs:choice in the schema but it's actually just us internally trying to parse two different things.

Reply:
Actually I think ChoiceParser is only used for choices now. The grammar no longer has the "|" operator in it at all, so there is no runtime alternation. All such conditional behavior is done by some explicit combinator. All the grammar alternatives must be resolvable at compilation time now, and use the "||" operator, which insists on turning into just one alternative at compile time.